pkg/services/dashboards/dashboard_service.go

```diff
@@ -14,14 +14,14 @@ import (
 
 // DashboardService service for operating on dashboards
 type DashboardService interface {
-	SaveDashboard(dto *SaveDashboardDTO) (*models.Dashboard, error)
+	SaveDashboard(dto *SaveDashboardDTO, allowUiUpdate bool) (*models.Dashboard, error)
 	ImportDashboard(dto *SaveDashboardDTO) (*models.Dashboard, error)
 	DeleteDashboard(dashboardId int64, orgId int64) error
 }
 
 // DashboardProvisioningService service for operating on provisioned dashboards
 type DashboardProvisioningService interface {
-	SaveProvisionedDashboard(dto *SaveDashboardDTO, provisioning *models.DashboardProvisioning) (*models.Dashboard, error)
+	SaveProvisionedDashboard(dto *SaveDashboardDTO, provisioning *models.DashboardProvisioning, allowUiUpdates bool) (*models.Dashboard, error)
```

Comment:
I think go vet is throwing here https://circleci.com/gh/grafana/grafana/169568?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link because you've changed the contract of this method without updating the `fakeDashboardProvisioningService` here https://github.com/grafana/grafana/blob/e6fa7c2e04602fabe26cd5dba9369451baa8edb0/pkg/services/provisioning/dashboards/file_reader_test.go#L371

Therefore the `fakeDashboardProvisioningService` doesn't no longer satisfies the `DashboardProvisioningService` interface. Makes sense?

Reply:
Absolutely, thanks!"